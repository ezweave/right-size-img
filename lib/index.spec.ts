import mkdirp from 'mkdirp';
import path from 'path';
import { rmdirSync } from 'fs';
import { resizeAllImagesInDirectory } from '.';

const fixturePath = path.resolve(
	__dirname,
	'./__fixtures__'
);

const tempDirectoryPath = 'tmp';
let tempDirectory;

describe('resizeAllImagesInDirectory', () => {
	beforeAll(async () => {
		tempDirectory = path.resolve(
			__dirname,
			tempDirectoryPath
		);
		await mkdirp(tempDirectory);
	});
	it('resize all images in a directory', async () => {
		const results = await resizeAllImagesInDirectory({
			maxWidth: 1500,
			pathToFiles: fixturePath,
			outputPath: tempDirectory,
		});
		expect(results).toMatchSnapshot();
	});
	afterAll(() => {
		rmdirSync(tempDirectory, { recursive: true });
	})
});