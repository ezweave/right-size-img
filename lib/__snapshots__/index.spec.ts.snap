// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resizeAllImagesInDirectory resize all images in a directory 1`] = `
Array [
  "/Users/mweaver/code/right-size-img/lib/__fixtures__/large_rectangle.png converted from 2000x1358 to 1500x1019, written to /Users/mweaver/code/right-size-img/lib/tmp/large_rectangle.png",
  "/Users/mweaver/code/right-size-img/lib/__fixtures__/large_square.png converted from 6000x6000 to 1500x1500, written to /Users/mweaver/code/right-size-img/lib/tmp/large_square.png",
  "/Users/mweaver/code/right-size-img/lib/__fixtures__/medium_square.png converted from 1500x1500 to 1500x1500, written to /Users/mweaver/code/right-size-img/lib/tmp/medium_square.png",
  "/Users/mweaver/code/right-size-img/lib/__fixtures__/small_rectangle.png too small for resize, copied to /Users/mweaver/code/right-size-img/lib/tmp/small_rectangle.png",
]
`;
